warden/x/async/keeper/keeper.go

```diff
@@ -360,48 +422,53 @@ func (k Keeper) getCompletedTasksWithoutValidatorVote(ctx context.Context, valAd
 	return tasks, nil
 }
 
-func (k *Keeper) AddPlugin(ctx context.Context, p types.Plugin) error {
-	id := p.GetId()
-
-	if id == "" {
-		return errors.New("plugin ID cannot be empty")
-	}
-
-	found, err := k.plugins.Has(ctx, id)
+func (k Keeper) releaseFee(ctx context.Context, id uint64, submitter []byte) error {
```

Comment:
let's
- take the Task directly instead of the id
- use sdk.ConsAddress instead of untyped []byte to make the intent clearer

```suggestion
func (k Keeper) releaseFee(ctx context.Context, task types.Task, submitter sdk.ConsAddress) error {
```